src/component/common/Constraint/Constraint.tsx

```diff
@@ -1,31 +1,77 @@
+import { Button } from '@material-ui/core';
 import classnames from 'classnames';
+import { useContext } from 'react';
+import AccessContext from '../../../contexts/AccessContext';
 import { IConstraint } from '../../../interfaces/strategy';
 import FeatureStrategiesSeparator from '../../feature/FeatureView2/FeatureStrategies/FeatureStrategiesEnvironments/FeatureStrategiesSeparator/FeatureStrategiesSeparator';
+import { UPDATE_FEATURE } from '../../providers/AccessProvider/permissions';
+import ConditionallyRender from '../ConditionallyRender';
 import StringTruncator from '../StringTruncator/StringTruncator';
 import { useStyles } from './Constraint.styles';
 
 interface IConstraintProps {
     constraint: IConstraint;
     className?: string;
+    deleteCallback?: () => void;
+    editCallback?: () => void;
 }
 
-const Constraint = ({ constraint, className, ...rest }: IConstraintProps) => {
+const Constraint = ({
+    constraint,
+    deleteCallback,
+    editCallback,
+    className,
+    ...rest
+}: IConstraintProps) => {
     const styles = useStyles();
+    const { hasAccess } = useContext(AccessContext);
 
     const classes = classnames(styles.constraint, {
         [styles.column]: constraint.values.length > 2,
     });
 
+    const containerClasses = classnames(
+        styles.constraint,
+        styles.constraintContainer,
+        {
+            [styles.column]: true,
+        }
+    );
+
+    const editable = deleteCallback && editCallback;
+
     return (
-        <div className={classes + ' ' + className} {...rest}>
-            <StringTruncator text={constraint.contextName} maxWidth="125" />
-            <FeatureStrategiesSeparator
-                text={constraint.operator}
-                maxWidth="none"
+        <div className={containerClasses + ' ' + className} {...rest}>
```

Comment:
you can import classnames and use that to join classes a little more smoothly: 

```
import classnames from 'classnames';

classnames(containerClasses, className)
```

Reply:
That's super neat, thanks for the tip! Oddly enough, I had to restructure the CSS so I no longer need it but I'm going to file that in the back of my mind for next time!